### PR TITLE
[JENKINS-39112] Move GitHubLibraryResolver to github branch source plugin

### PR DESCRIPTION
**Pull request by stephenc**
See [JENKINS-39112](https://issues.jenkins-ci.org/browse/JENKINS-39112)

@reviewbybees

### CODE REVIEW

**Comment by ghost** [comment]
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

**Review by jglick — APPROVED** [review]
-0, see linked PR

**Comment by rsandell** [comment]
:bee:

**Comment by stephenc** [comment]
@jglick ping

File: pom.xml [hunk 1]
```diff
@@ -69,6 +69,11 @@
       <version>1.11-SNAPSHOT</version>
     </dependency>
     <dependency>
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>pipeline-github-lib</artifactId>
+      <version>1.0-SNAPSHOT</version>
```
Comment (jglick) [inline pom.xml:74]: There is no “creation PR” there, so leaving feedback here:

That plugin does not really need a dep on `github-branch-source`. It only actually uses `git`, at least currently.

**Review by jglick — APPROVED** [review]
:bee:

**Pull request merged**